jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/ThreadMXBeanImpl.java

```diff
@@ -255,7 +267,21 @@ public ThreadInfo getThreadInfo(long id) {
 		}
 
 		// Create an array and populate with individual ThreadInfos
-		return this.getMultiThreadInfoImpl(ids, maxDepth, false, false);
+		ThreadInfoBase[] infoBases = this.getMultiThreadInfoImpl(ids, maxDepth, false, false);
+		return wrapBases(infoBases);
+	}
+
+	/**
+	 * Create ThreadInfo objects containing ThreadInfoBase objects.
+	 * @param infoBases containers for the actual thread information
+	 * @return ThreadInfo array whose elements wrap the elements of infoBases 
+	 */
+	private static ThreadInfo[] wrapBases(ThreadInfoBase[] infoBases) {
+		ThreadInfo[] result = Arrays.stream(infoBases)
+				.map(b -> makeThreadInfo(b))
+				.collect(Collectors.toList())
+				.toArray(THREAD_INFO_EMPTY_ARRAY);
```

Comment:
If we're already using lambdas here, I suggest `.collect().toArray()` should be replaced by
```
	.toArray(ThreadInfo[]::new);
```

Reply:
Thanks for the tip.